yt_dlp/extractor/mave.py

```diff
@@ -68,40 +117,75 @@ class MaveIE(InfoExtractor):
             'upload_date': '20241230',
         },
     }]
-    _API_BASE_URL = 'https://api.mave.digital/'
 
     def _real_extract(self, url):
-        channel_id, slug = self._match_valid_url(url).group('channel', 'id')
-        display_id = f'{channel_id}-{slug}'
-        webpage = self._download_webpage(url, display_id)
-        data = traverse_obj(
-            self._search_nuxt_json(webpage, display_id),
-            ('data', lambda _, v: v['activeEpisodeData'], any, {require('podcast data')}))
+        channel_id, episode_code = self._match_valid_url(url).group(
+            'channel_id', 'episode_code')
+
+        channel_meta = self._load_channel_meta(channel_id)
+        episode_meta = self._load_episode_meta(channel_id, episode_code)
+
+        return self._create_entry(channel_id, channel_meta, episode_meta)
+
+
+class MaveChannelIE(MaveBaseIE):
+    IE_NAME = 'mave:channel'
+    _VALID_URL = r'https?://(?P<id>[\w-]+)\.mave\.digital/?(?:$|[?#])'
+    _TESTS = [{
+        'url': 'https://budem.mave.digital/',
+        'info_dict': {
+            'id': 'budem',
+            'title': 'Все там будем',
+            'description': 'md5:f04ae12a42be0f1d765c5e326b41987a',
+        },
+        'playlist_mincount': 15,
+    }, {
+        'url': 'https://ochenlichnoe.mave.digital/',
+        'info_dict': {
+            'id': 'ochenlichnoe',
+            'title': 'Очень личное',
+            'description': 'md5:ee36a6a52546b91b487fe08c552fdbb2',
+        },
+        'playlist_mincount': 20,
+    }, {
+        'url': 'https://geekcity.mave.digital/',
+        'info_dict': {
+            'id': 'geekcity',
+            'title': 'Мужчины в трико',
+            'description': 'md5:4164d425d60a0d97abdce9d1f6f8e049',
+        },
+        'playlist_mincount': 80,
+    }]
+
+    def _real_extract(self, url):
+        channel_id = self._match_id(url)
+
+        channel_meta = self._load_channel_meta(channel_id)
+
+        entries = []
+        page = 1
+
+        while True:
+            data = self._download_json(
+                f'{self._API_BASE_URL}/{channel_id}/episodes?'
+                f'view=all&page={page}&sort=newest&format=all',
+                channel_id)
+
+            episodes = data.get('episodes', [])
+
+            if not episodes:
+                break
+            else:
+                for episode_meta in data['episodes']:
+                    entries.append(self._create_entry(
+                        channel_id, channel_meta, episode_meta))
+
+                page += 1
```

Comment:
I would use an InAdvancePagedList here
```suggestion
    _PAGE_SIZE = 50

    def _entries(self, channel_id, channel_meta, page_num):
        page_data = self._download_json(
            f'{self._API_BASE_URL}/{channel_id}/episodes', channel_id, query={
                'view': 'all',
                'page': page_num + 1,
                'sort': 'newest',
                'format': 'all',
            }, note=f'Downloading page {page_num + 1}')
        for ep in traverse_obj(page_data, ('episodes', ..., {dict})):
            yield self._create_entry(channel_id, channel_meta, episode_meta=ep)

    def _real_extract(self, url):
        channel_id = self._match_id(url)

        channel_meta = self._load_channel_meta(channel_id)
```